dask_ml/utils.py

```diff
@@ -241,8 +241,8 @@ def check_random_state(random_state):
         raise TypeError("Unexpected type '{}'".format(type(random_state)))
 
 
-def check_matching_blocks(*arrays):
-    """Check that the partitioning structure for many arrays matches.
+def _check_matching_blocks(*arrays, check_first_dim_only=False):
```

Comment:
I thought this was the best way to re-use code between check_consistent_length and check_matching_blocks without changing the api of either function.